eng/Directory.Build.Common.props

```diff
@@ -157,6 +158,10 @@
     <PackageReference Include="Azure.ResourceManager" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(IsProvisioningLibrary)' == 'true'">
```

Comment:
I really hate the approach of injecting implicit references in the build process rather than having them explicit.   This pattern creates confusion because references just appear from nowhere.

Unless there's a scenario that strongly justifies this, I'm going to ask that we just add the reference in the packages that need it.

//cc: @ArthurMa1978